js/client-mainmenu.js

```diff
@@ -114,7 +131,118 @@
 				);
 			}
 		},
+		createLoginFormObserver:function(self){
+			console.log(window.Storage.prefs("page"))
+			const config = { attributes: true, childList: true, subtree: true };
+			const observer = new MutationObserver(function(){
+			if(window.app.user.get("named") && window.Storage.prefs("page")=="challenge"){
+				window.Storage.prefs("page","testclient");
+				if($('.challenge-container').length==0){
+					//user log in event trigerd when user log in and make user pick a team
+					$('body').on("login_event",function(){
+					self.onUserLogIn(self);
+				})
+				$('body').trigger("login_event");
```

Comment:
`app.trigger` is probably better than adding the event to the DOM.
Also, prefer `login` over `login_event`.